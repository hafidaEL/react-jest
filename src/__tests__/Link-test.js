
import React from 'react';
import Link from '../Link';
import renderer from 'react-test-renderer';


it('rendu correct du composant Link ', () => {
  const tree = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('rendu sans prop page ', () => {
  const tree = renderer.create(
    <Link>Facebook</Link>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});



/*
test('Le lien change de classe au hover', () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">Facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}); */


