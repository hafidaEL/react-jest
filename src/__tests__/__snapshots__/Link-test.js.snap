exports[`test changes the class when hovered 1`] = `
<div>
  le lien : 
  <a
    className="normal"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    Facebook
  </a>
</div>
`;

exports[`test properly escapes quotes 1`] = `
<div>
  le lien : 
  <a
    className="normal"
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    \"Facebook\" \\\'is \\ \'awesome\'
  </a>
</div>
`;

exports[`test renders as an anchor when no page is set 1`] = `
<div>
  le lien : 
  <a
    className="normal"
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    Facebook
  </a>
</div>
`;

exports[`test rendu correct du composant Link  1`] = `
<div>
  le lien : 
  <a
    className="normal"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    Facebook
  </a>
</div>
`;

exports[`test rendu sans prop page  1`] = `
<div>
  le lien : 
  <a
    className="normal"
    href="#"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    Facebook
  </a>
</div>
`;
